### PR TITLE
Review

### PR DESCRIPTION
**Pull request by jo-pol**
Just a partial review.

Examine the commits and their comments one by one to learn. Somme commit comments are marked with TODO, you can apply the same principle somewhere else.

Somewhere I threw a new exception. The message might need to be different or constructed at a different level. This is a functional change not marked as such in the commit comment. Please add tests that show these error messages.

I see you frequently create a buffer and then add elements in a `forearch`, that can usually be replaced by changing the `foreach` into a `map`. In other situations you create some collection an then add elements with `+=`. In those situations you can usually use a var-arch constructor. You can find various some in the commits and apply them in other situations.

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.s2d/dataverse/HttpSupport.scala [hunk 1]
```diff
@@ -86,47 +87,48 @@ trait HttpSupport extends DebugEnhancedLogging {
...
       _ = debug(s"Request URL = $uri")
-      response <- http("DELETE", uri, null, Map("X-Dataverse-key" -> apiToken))
+      response <- http("DELETE", uri, null, Map(tokenHeader))
       _ <- handleResponse(response, 200)
     } yield s"Successfully DELETED: $uri"
   }
 
   protected def tryReadFileToString(file: File): Try[String] = Try {
-    FileUtils.readFileToString(file.toJava, StandardCharsets.UTF_8)
+    file.contentAsString(StandardCharsets.UTF_8)
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.s2d/dataverse/HttpSupport.scala:125]: This method has nothing to do with HttpSupport, it belongs in a package object.

File: src/main/scala/nl.knaw.dans.easy.s2d/dataverse/Dataset.scala [hunk 2]
```diff
@@ -23,7 +23,7 @@ import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 
 import scala.util.{ Failure, Try }
 
-class Dataset(id: String, isPersistentId: Boolean, configuration: DataverseInstanceConfig)(implicit resultOutput: PrintStream) extends HttpSupport with DebugEnhancedLogging {
+case class Dataset(id: String, isPersistentId: Boolean, configuration: DataverseInstanceConfig)(implicit resultOutput: PrintStream) extends HttpSupport with DebugEnhancedLogging {
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.s2d/dataverse/Dataset.scala:26]: For a testable version return the case class to an ordinary class and:

```
class Dataset(id: String, isPersistentId: Boolean, configuration: DataverseInstanceConfig)(implicit resultOutput: PrintStream) extends DebugEnhancedLogging {
  protected lazy val httpSupport = new HttpSupport {
    override protected val connectionTimeout: Int = configuration.connectionTimeout
    override protected val readTimeout: Int = configuration.readTimeout
    override protected val baseUrl: URI = configuration.baseUrl
    override protected val apiToken: String = configuration.apiToken
    override protected val apiVersion: String = configuration.apiVersion
  }
```

Your test class then can become (I hope) something like 

```
class DatasetSpec extends FlatSpec with Matchers with MockFactory{

  class MockedHttpSupport extends HttpSupport {
    val apiToken: String = null
    val connectionTimeout: Int = 0
    val apiVersion: String = null
    val baseUrl: URI = null
    val readTimeout: Int = null
  }
  class MockedDataset(id: String, isPersistentId: Boolean, configuration: DataverseInstanceConfig)(implicit resultOutput: PrintStream)
    extends Dataset(id, isPersistentId, configuration)(resultOutput) {
    override lazy val httpSupport: HttpSupport = mock[HttpSupport]
  }
```

File: src/test/scala/nl.knaw.dans.easy.s2d/EasyToDataverseMapperSpec.scala [hunk 3]
```diff
@@ -21,8 +21,8 @@ import org.scalatest.{ FlatSpec, Matchers, OneInstancePerTest }
 
 class EasyToDataverseMapperSpec extends FlatSpec with OneInstancePerTest with Matchers {
 
-  implicit val format = DefaultFormats
-  val mapper = new EasyToDataverseMapper
+  private implicit val format: DefaultFormats.type = DefaultFormats
+  private val mapper = EasyToDataverseMapper()
```
Comment (jo-pol) [inline src/test/scala/nl.knaw.dans.easy.s2d/EasyToDataverseMapperSpec.scala:25]: Just for fun I tried the following. It actually produced a lot of empty fields. Is that intentionally?

```scala
  "mapToJson" should "..." in {
    mapper.mapToJson(
       <ddm:DDM>
            <ddm:profile>
              <ddm:title>Agricultural Sciences</ddm:title>
            </ddm:profile>
         </ddm:DDM>
    ) shouldBe
      """"{
        |  "datasetVersion":{
        |    "metadataBlocks":{
        |      "citation":{
        |        "displayName":"Citation Metadata",
        |        "fields":[
        |          {
        |            "typeName":"title",
        |            "multiple":false,
        |            "typeClass":"primitive",
        |            "value":"Agricultural Sciences"
        |          }
        |        ]
        |      }
        |    }
        |  }
        |}"""".stripMargin
  }
```

**Comment by janvanmansum** [comment]
Closing this one.

**Pull request closed without merging**